src/server_manager/web_app/app.ts

```diff
@@ -505,13 +505,16 @@ export class App {
     };
     const oauthUi = this.appRoot.getDigitalOceanOauthFlow(signOutAction);
     for (;;) {
-      const status = await this.digitalOceanRetry(async () => {
+      const info = await this.digitalOceanRetry(async () => {
         if (cancelled) {
           throw CANCELLED_ERROR;
         }
         return await digitalOceanAccount.getStatus();
       });
-      if (status === digitalocean.Status.ACTIVE) {
+      if (info.warning) {
+        this.appRoot.showError(`DigitalOcean warning: ${info.warning}`);
+      }
+      if (info.status === digitalocean.Status.ACTIVE) {
```

Comment:
I think we should show an error if the account is locked.

Reply:
I'm not sure I understand.  If the account status is locked, and it has no droplets, it will be marked as "needsEmailVerification" or "needsBillingInfo", leading to an interstitial.  Otherwise, login will succeed, and any warning message will be displayed.  I assume that if the account is locked, DO always includes a warning message.